### ISSUE TITLE
In new Annotations module, shapes cut off at edges of line chart

### ISSUE THREAD

**Issue by rachelfreya** [highcharts/highcharts#7861]
#### Expected behaviour
When annotations are added using the new module, chart will automatically adjust to fit shapes attached to the first or last point of a line chart.

#### Actual behaviour
Shapes are cut off.

#### Live demo with steps to reproduce
http://jsfiddle.net/1zm3ut2w/

#### Product version
Highstock 6.0.4

#### Affected browser(s)
All

**Comment by TorsteinHonsi** [comment]
Thanks for your suggestion! Detecting space needed for the annotations is potentially quite complex, so for you you are required to handle this manually.

One way would be to make sure the axes are long enough http://jsfiddle.net/1zm3ut2w/2/. Another way would be not to clip the `shapesGroup`, but that requires a hack: http://jsfiddle.net/1zm3ut2w/7/.

@morganfree Shouldn't the shapes support the `crop` option like labels? Or some kind of clip option?

**Comment by morganfree** [comment]
@TorsteinHonsi, Yes, I think it should be supported. I think we used optional clipping in the advanced Highstock 6 demo - it could be set in an annotation options and it allowed to choose a specific clip.

**Comment by AmolKedari1492** [comment]
I was facing same, I was using multiple series and annotation shapes were getting clipped but marker were not. So what I did was nothing but unsetting clip-path of class highcharts-annotation-shapes. I think it will work here also.

**Comment by pawelfus** [comment]
Unstale

**Comment by MrR341-GER** [comment]
I want to upvote this!

I made a workaround playing with the hichchart.js-code deleting the part where he enters an clip-part="url('#highchart-someId-')"
